specification/0002-quilt.mod.json.md

```diff
@@ -398,9 +398,32 @@ A mod identifier in the form of either `mavenGroup:modId` or `modId`.
 ### The `versions` field
 | Type         | Required | Default |
 |--------------|----------|---------|
-| Array/String | False    | `"*"`   |
+| Array/Object/String | False    | `"*"`   |
 
-Should be a [version specifier](#version-specifier) or array of version specifiers defining what versions this dependency applies to. If an array of versions is provided, the dependency matches if it matches ANY of the listed versions.
+A version specifier, or complex set of version specifiers that control what versions match this dependency object.
+
+It is an error to specify multiple constraints that conflict with each other (where no version would match the whole specifier).
+
+It is an error if the resulting version specifier matches every version, and isn't the single string `"*"`, or uses the deprecated array style of definition.
+
+#### String
+
+Should be a single [version specifier](#version-specifier) defining the versions this dependency applies to.
+
+#### Array (Deprecated)
+
+Should be an array of [version specifiers](#version-specifier) defining the versions this dependency applies to. The dependency matches if it matches ANY of the listed versions.
+
+This usage is discouraged as many mod developers wrongly assumed this was "ALL" rather than "ANY", leading to unfortunate situations where mods depend on [ ">=1.18.1", "<1.19.3" ], which actually matches every version of minecraft, rather than just the versions between 1.18.1 and 1.19.2.
```

Comment:
```suggestion
This usage is discouraged as many mod developers wrongly assumed this was "ALL" rather than "ANY", leading to unfortunate situations where mods depend on [ ">=1.18.1", "<1.19.3" ], which actually matches _every_ version of Minecraft, rather than just the versions between 1.18.1 and 1.19.2.
```